### PR TITLE
Feature/game state/inter planetary travel

### PR DESCRIPTION
**Pull request by VuKhanhVyHo**
This pull allows travel between planets and spawn the specified minigame screen for each round while moving between planets. A JUnit test to test the class is also attached to ensure the state changed after call the method

### CODE REVIEW

File: source/core/src/main/com/csse3200/game/services/PlanetTravel.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+         * The current gameState
+         */
+        private final GameState gameState;
+
+        /**
+         * Constructor to set the current game state for using
+         * @param gameState
+         */
+        public PlanetTravel(GameState gameState, GdxGame game) {
+            this.gameState = gameState;
```
Comment (henrybatt) [inline source/core/src/main/com/csse3200/game/services/PlanetTravel.java]: Incorrect accessing of the global game state, can you please fix this.

File: source/core/src/test/com/csse3200/game/services/PlanetTravelTest.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    private GdxGame mockGame; // Mock GdxGame
+
+    @BeforeEach
+    public void setUp() {
+        InputFactory inputFactoryMock = mock(InputFactory.class);
+        Input inputMock = mock(Input.class);
+        Gdx.input = inputMock;
+        mockGame = mock(GdxGame.class);
+        InputService inputService = new InputService(inputFactoryMock);
+        gameState = new GameState();
```
Comment (henrybatt) [inline source/core/src/test/com/csse3200/game/services/PlanetTravelTest.java]: This JUnit test might need to be rewritten depending on changes made to update GameState appropriately

**Review by henrybatt — CHANGES_REQUESTED** [review]
Can you please ensure you are accessing the game state appropriately

**Comment by TDay1** [comment]
Looks good!

**Review by henrybatt — APPROVED** [review]
Looks good, thanks.

**Pull request merged**